include/platform/mir/graphics/renderable.h

```diff
@@ -69,6 +72,9 @@ class Renderable
     virtual glm::mat4 transformation() const = 0;
 
     virtual bool shaped() const = 0;  // meaning the pixel format has alpha
+
+    virtual auto surface_if_any() const
+        -> std::optional<mir::scene::Surface const*> = 0;
```

Comment:
`std::optional<pointer const*>` is not *super* different to plain `pointer const*`; I could go either way on this; we should probably have an actual style guide decision on this.

Reply:
So long as we adhere to the Core Guidelines:

> R.2: In interfaces, use raw pointers to denote individual objects (only)
> R.3: A raw pointer (a T*) is non-owning

I'm also fine with a named `Foo cont*` provided that there's a clear indication (like `_if_any`) when `nullptr` is possible.

For reference, Sophie was keen on `std::optional<>`, so there's some of that around the codebase.

Reply:
I like `std::optional` personally because it shows intent. We don't `null`-check every `shared_ptr` in the repo, so I would assume that when we're accessing _any_ pointer, we can assume that it is probably set.